x-pack/legacy/plugins/ml/public/datavisualizer/datavisualizer_controller.js

```diff
@@ -144,12 +145,21 @@ module
       .value();
     $scope.indexedFieldTypes = indexedFieldTypes.sort();
 
-
-    // Refresh the data when the time range is altered.
-    $scope.$listenAndDigestAsync(timefilter, 'fetch', function () {
+    function refresh() {
       $scope.earliest = timefilter.getActiveBounds().min.valueOf();
       $scope.latest = timefilter.getActiveBounds().max.valueOf();
       loadOverallStats();
+    }
+
+    // Refresh the data when the time range is altered.
+    $scope.$listenAndDigestAsync(timefilter, 'fetch', function () {
```

Comment:
could be arrow function, or even just 
`$scope.$listenAndDigestAsync(timefilter, 'fetch', refresh);`

Reply:
Will address this and the other suggestions in the follow-up 👍